QuantConnect.DataBento/DataBentoDataProvider.cs

```
@@ -112,121 +86,69 @@ public DataBentoProvider(string apiKey)
-                                            }
-                                        }
-                                    }
-                                }
-                                else
-                                {
-                                    Log.Error($"DataBentoProvider.SubscribeImpl(): Failed to subscribe to live data for {config.Symbol}");
-                                }
-                            });
+                            Log.Error($"DataBentoProvider.SubscribeImpl(): Failed to subscribe to live data for {config.Symbol}");
+                            return;
                         }
-                        else
+
+                        // Start session once after first successful subscription
+                        lock (_sessionLock)
                         {
-                            Log.Trace($"DataBentoProvider.SubscribeImpl(): Client not connected, skipping subscription for {symbol}");
+                            if (!_sessionStarted)
+                            {
+                                _sessionStarted = _client.StartSession();
+                            }
                         }
-                    }
-                    else
-                    {
-                        Log.Trace($"DataBentoProvider.SubscribeImpl(): No config found for {symbol}, skipping");
-                    }
+                    });
                 }
-
                 return true;
             };
 
             _subscriptionManager.UnsubscribeImpl = (symbols, tickType) =>
             {
-                Log.Trace($"DataBentoProvider.UnsubscribeImpl(): Received unsubscribe request for {symbols.Count()} symbols, TickType={tickType}");
-
                 foreach (var symbol in symbols)
                 {
-                    Log.Trace($"DataBentoProvider.UnsubscribeImpl(): Processing symbol {symbol}");
-
                     if (_client?.IsConnected == true)
```

this would silently fail to subscribe if not connected; you should either throw or try to reconnect

---

Private comment was to invert the If to reduce nesting.